rfc/produce-message-size.md

```diff
@@ -0,0 +1,99 @@
+# Enhancing Fluvio to Message Sizes
+
+This RFC proposes modifications to Fluvio's handling of message sizes. 
+
+## Proposed Enhancements
+
+1. Handling Larger Messages than Batch Size
+
+If a single record exceeds the defined `batch_size`, Fluvio will process the record as a standalone request, ensuring that larger messages are not discarded or delayed. If the record does not exceed the `batch_size`, Fluvio will process the record as part of an already existing batch or create a new one if the batch is full.
+
+2. Handling Larger Messages than the Max Request Size
+
+Fluvio will have a new configuration parameter, max_request_size, that will define the maximum size of a request that can be sent by the producer. This configuration will make Fluvio display errors when a message exceeds the defined `max_request_size`, even if it's a message with only one record or a batch of records.
+
+3. Compression Behavior
+
+Fluvio will ensure that configuration limits that use size constraints, such as `batch_size` and `max_request_size` will only use the uncompressed message size.
+It affects the size of messages in transit but doesn't change the maximum request size constraints.
+
+
+## Fluvio CLI
+
+Preparing the environment, with a topic and a large data file:
+
+```bash
+fluvio topic create large-data-topic
+printf 'This is a sample line. ' | awk -v b=500000 '{while(length($0) < b) $0 = $0 $0}1' | cut -c1-500000 > large-data-file.txt
+```
+
+### Batch Size
+
+`batch_size` will define the maximum size of a batch of records that can be sent by the producer. If a record exceeds this size, Fluvio will process the record as a standalone message.
```

Comment:
`batch_size` should be ultimate raw limit.   again not sure what this address

Reply:
I think this behavior will be confused, other products has the same parameter config `batch_size`. In other products, Its behavior is to just check if the record will be inside a batch or not.

Reply:
not reject them